packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1913,8 +1939,18 @@ abstract class _NumProperty<T extends num> extends DiagnosticsProperty<T> {
   }) : super.lazy();
 
   @override
-  Map<String, Object?> toJsonMap(DiagnosticsSerializationDelegate delegate) {
-    final Map<String, Object?> json = super.toJsonMap(delegate);
+  Map<String, Object?> toJsonMap(
+    DiagnosticsSerializationDelegate delegate, {
+    bool fullDetails = true,
+  }) {
+    final Map<String, Object?> json = super.toJsonMap(
+      delegate,
+      fullDetails: fullDetails,
+    );
+    if (!fullDetails) {
```

Comment:
nit: same here, this could be reduced a bit by doing
```
if (fullDetails) {
  if (unit != null) {
    json['unit'] = unit;
  }
  json['numberToString'] = numberToString();
}
return json;
```

same comment for others below. If you think it is better to return early feel free to push back.

Reply:
I prefer returning early to avoid nesting the conditionals, leaving as is